rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/ChargingRSpace.scala

```diff
@@ -39,14 +41,24 @@ object ChargingRSpace {
           patterns: Seq[BindPattern],
           continuation: TaggedContinuation,
           persist: Boolean,
-          sequenceNumber: Int
+          sequenceNumber: Int,
+          peek: Boolean
       ): F[
         Option[(ContResult[Par, BindPattern, TaggedContinuation], Seq[Result[ListParWithRandom]])]
       ] =
         for {
-          _       <- charge[F](storageCostConsume(channels, patterns, continuation))
-          consRes <- space.consume(channels, patterns, continuation, persist, sequenceNumber)
-          _       <- handleResult(consRes)
+          _ <- charge[F](storageCostConsume(channels, patterns, continuation))
+          consRes <- space.consume(
+                      channels,
+                      patterns,
+                      continuation,
+                      persist,
+                      sequenceNumber,
+                      if (peek) {
```

Comment:
nitpick: don't you think that an if merits a separate line in the for?

Reply:
Can do